core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -803,6 +864,14 @@ private[log] class Cleaner(val id: Int,
     growBuffers(maxSize)
   }
 
+  /**
+   * Judge a batch should be discard by cleaned transaction state
```

Comment:
I find "Judge" a bit odd in this situation.
```suggestion
   * Determine if a batch should be discard by cleaned transaction state
```

Reply:
You could also use "check" instead of "determine"